pkg/releaselock/rpc.go

```diff
@@ -86,23 +88,31 @@ func ReleaseLock(hostIP, clientIP string) error {
 	}()
 
 	// Create client using sunrpc codec.
-	client := rpc.NewClientWithCodec(sunrpc.NewClientCodec(conn, notifyClose))
+	client := sunrpc.NewClientCodec(conn, notifyClose)
 
 	klog.Infof("Calling Filestore IP %s to release all locks for GKE node %s", serverAddress, clientIP)
 
-	ipByte := net.ParseIP(clientIP).To4()
-	if ipByte == nil {
+	ip := net.ParseIP(clientIP)
+	if ip == nil {
 		return fmt.Errorf("invalid GKE node IP %s", clientIP)
 	}
+	ipByte := ip.To4()
+	if ipByte == nil {
+		return fmt.Errorf("invalid GKE node IPv4 address %s", clientIP)
+	}
 	ipBinary := binary.BigEndian.Uint32(ipByte)
 
-	var response releaseLockResponse
-	if err := client.Call(inbandLockReleaseProcedureName, ipBinary, response /* empty response body */); err != nil {
-		return fmt.Errorf("failed to call procedure %s for GKE node IP %s Filestore IP %s, err: %w", inbandLockReleaseProcedureName, clientIP, hostIP, err)
+	request := rpc.Request{
+		ServiceMethod: inbandLockReleaseProcedureName,
+		Seq:           rand.New(rand.NewSource(time.Now().UnixNano())).Uint64(),
```

Comment:
I would recommend not drawing a random number every time, but initiate a variable with a random value. This way you don't have the small chance of getting the same XID twice.

Reply:
Done